### ISSUE TITLE
Dropout applied twice before LSTM inputs

### ISSUE THREAD

**Issue by samgd** [mozilla/DeepSpeech#1201]
As per my note on mozilla/#machinelearning:

`BiRNN` applies dropout after the third layer:

```
    layer_3 = tf.nn.dropout(layer_3, (1.0 - dropout[2]))
```

`layer_3` is then used as input to the bidirectional rnn:

```
    outputs, output_states = tf.nn.bidirectional_dynamic_rnn(cell_fw=lstm_fw_cell,
                                                             cell_bw=lstm_bw_cell,
                                                             inputs=layer_3,
                                                             dtype=tf.float32,
                                                             time_major=True,
                                                             sequence_length=seq_length)
```

However both `lstm_fw_cell` and `lstm_bw_cell` have a: 

```
    lstm_fw_cell = tf.contrib.rnn.DropoutWrapper(lstm_fw_cell,
                                                input_keep_prob=1.0 - dropout[3],
                                                output_keep_prob=1.0 - dropout[3],
                                                seed=FLAGS.random_seed)
```

```
    lstm_bw_cell = tf.contrib.rnn.DropoutWrapper(lstm_bw_cell,
                                                input_keep_prob=1.0 - dropout[4],
                                                output_keep_prob=1.0 - dropout[4],
                                                seed=FLAGS.random_seed)
```

which applies dropout to the inputs also meaning that dropout will be applied again to `layer_3`. 

As noted by gerard-majax `dropout_rate[4]` is by default set to 0.0 so this likely only effects `lstm_fw_cell` which defaults to `dropout_rate` unless the hyperparameters have been changed otherwise.

I believe this results in an effective `keep_prob=(1.0 - dropout[2]) * (1.0 - dropout[3])` for `lstm_fw_cell`.

**Comment by kdavis-mozilla** [comment]
`dropout_rate4` and `dropout_rate5` default to 0, see [DeepSpeech.py#L67](https://github.com/mozilla/DeepSpeech/blob/master/DeepSpeech.py#L67) and [DeepSpeech.py#L68](https://github.com/mozilla/DeepSpeech/blob/master/DeepSpeech.py#L68).

This implies `dropout_rates[3]` and `dropout_rates[4]` default to 0, see [DeepSpeech.py#L222](https://github.com/mozilla/DeepSpeech/blob/master/DeepSpeech.py#L222).

This in turn implies `dropout[3]` and `dropout[4]` above default to 0.
